src/engine/engine_wb.c

```diff
@@ -109,6 +109,8 @@ static void _ocf_write_wb_complete(struct ocf_request *req, int error)
 
 		ocf_engine_invalidate(req);
 	} else {
+		_ocf_write_wb_update_bits(req);
```

Comment:
sorry, you should actually move it to ocf_write_wb_do_flush_metadata - updating status bits acquires a mutex and here we are potentially in an atomic context.